woocommerce/woocommerce
# Don't recommend WCS&T in tax task if WCShip or WCTax is active

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Part of woocommerce/woocommerce-shipping#187.

If the "WooCommerce Shipping" or "WooCommerce Tax" extension is active, don't recommend the incompatible "WooCommerce Shipping & Tax" extension.

This PR implements the above restriction in the **WC > Home > task list > Collect sales tax** task.

### Screenshots

Before:

(Note the screenshot recommends "WooCommerce Tax" but the solution used under the hood is actually the "WooCommerce Shipping & Tax" extension.)

<img width="740" alt="Screenshot 2024-06-06 at 13 59 50" src="https://github.com/woocommerce/woocommerce/assets/1759681/0132a9e6-2ca5-4c72-b364-51c1909439d5">

After, if WCShip or WCTax is present:

<img width="742" alt="Screenshot 2024-06-06 at 14 00 11" src="https://github.com/woocommerce/woocommerce/assets/1759681/5ff7489e-843f-4bc9-a9d4-495ac809f41e">

(The "Configure" button leads to the WC > Settings > Tax > Standard rates page.)



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Build WC from this PR and install it on a fresh site.
2. Skip the onboarding and set the store location to US California.
3. Go to **WC > Home** and try going through the **Collect sales tax** task with WCShip/WCTax enabled and disabled. WCS&T should only be recommended when neither of WCShip/WCTax is active.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

If either the upcoming WooCommerce Shipping or WooCommerce Tax extension is active, don't suggest WooCommerce Shipping & Tax in the "Collect sales tax" Home task.

This only adjusts the extension recommendation merchants receive and does not affect any other functionality.

</details>
